xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-rest/src/main/java/org/xwiki/search/solr/internal/rest/SOLRSearchSource.java

```diff
@@ -181,20 +180,27 @@
                     searchResult.setAuthorName((String) document.get(FieldUtils.AUTHOR_DISPLAY));
                 }
 
-                Locale locale = LocaleUtils.toLocale((String) document.get(FieldUtils.DOCUMENT_LOCALE));
+                Locale docLocale = LocaleUtils.toLocale((String) document.get(FieldUtils.DOCUMENT_LOCALE));
+                Locale locale = LocaleUtils.toLocale((String) document.get(FieldUtils.LOCALE));
+
+                List<String> titles = (List<String>) document.get(FieldUtils.getFieldName(FieldUtils.TITLE, locale));
```

Comment:
You should use ``document.getFirstValue(...)`` instead of doing the tests yourself below.

Reply:
Better, yes. Thanks.